cmd/opm/alpha/veneer/semver.go

```diff
@@ -0,0 +1,115 @@
+package veneer
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	// "github.com/blang/semver/v4"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	veneer_semver "github.com/operator-framework/operator-registry/alpha/veneer/semver"
+	"github.com/operator-framework/operator-registry/cmd/opm/internal/util"
+	containerd "github.com/operator-framework/operator-registry/pkg/image/containerdregistry"
+	"github.com/spf13/cobra"
+)
+
+func newSemverCmd() *cobra.Command {
+	output := ""
+	// templates := []string{}
+	skipPatch := false
+	// semverRangeStr := ""
+	minorChannels := false
+	majorChannels := false
+	cmd := &cobra.Command{
+		Use:   "semver <filename>",
+		Short: "Generate a declarative config blob from a single 'semver veneer' file",
+		Long:  `Generate a declarative config blob from a single 'semver veneer' file`,
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ref := args[0]
+
+			var (
+				// semverRange semver.Range
+				err error
+			)
+			// if semverRangeStr == "" {
+			// 	semverRange = func(semver.Version) bool { return true }
+			// } else {
+			// 	semverRange, err = semver.ParseRange(semverRangeStr)
+			// 	if err != nil {
+			// 		return fmt.Errorf("invalid semver range %q", semverRangeStr)
+			// 	}
+			// }
+
+			var write func(declcfg.DeclarativeConfig, io.Writer) error
+			switch output {
+			case "json":
+				write = declcfg.WriteJSON
+			case "yaml":
+				write = declcfg.WriteYAML
+			default:
+				return fmt.Errorf("invalid output format %q", output)
+			}
+
+			// The bundle loading impl is somewhat verbose, even on the happy path,
+			// so discard all logrus default logger logs. Any important failures will be
+			// returned from veneer.Render and logged as fatal errors.
+			logrus.SetOutput(ioutil.Discard)
+
+			skipTLSVerify, useHTTP, err := util.GetTLSOptions(cmd)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			cacheDir, err := os.MkdirTemp("", "veneer-registry-")
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			reg, err := containerd.NewRegistry(
+				containerd.WithCacheDir(cacheDir),
+				containerd.SkipTLSVerify(skipTLSVerify),
+				containerd.WithPlainHTTP(useHTTP),
+				containerd.WithLog(nullLogger()),
+			)
+			if err != nil {
+				log.Fatalf("creating containerd registry: %v", err)
+			}
+			defer reg.Destroy()
+
+			veneer := veneer_semver.Veneer{
+				Ref:           ref,
+				SkipPatch:     skipPatch,
+				ChannelsMajor: majorChannels,
+				ChannelsMinor: minorChannels,
```

Comment:
My opinion: we should make major channels optional, but minor channels always on. Production users will want to have assurances they can be on a channel that receives only back-compat patches.

Reply:
I agree, but the only thing which prevented me from taking this position is trying to anticipate the UX where only the major channels are desired, and the fact that it would make the flags conflict (e.g. --no-minor-channels, --major-channels). Instead of many variations on the "no on one, yes on the other" I felt it was most consistent if both were explicitly requested by the user.

Reply:
For the moment (in my local), I've converted these flags to 'omitempty' fields in the veneer itself, and provided a default of (major: false | minor: true | skips: true) for the values.  
Any deviation from default would then be specifically overridden by user-provided values in the veneer, and transparent since it's in the veneer.
How does that sound?

Reply:
Sounds perfect for the "let's get feedback phase" for sure. 👍